src/lib.rs

```diff
@@ -151,10 +155,31 @@ impl Tracer {
         Ok(String::from(map_filename))
     }
 
+    fn check_perf_perms() -> Result<(), TraceMeError> {
+        if unsafe { geteuid() } == 0 {
+            // Root can always trace.
+            return Ok(());
+        }
+
+        let mut f = File::open(&PERF_PERMS_PATH)?;
+        let mut buf = String::new();
+        f.read_to_string(&mut buf)?;
+        let buf = buf.trim();
+        let perm = buf.parse::<i8>()?;
```

Comment:
What's the use of having `perm` is an intrmediate variable? Wouldn't `if buf.trim().parse::<i8>()? != -1 { return Err(...) }` be simpler?

Reply:
Can do, yeah.